libnmstate/nm/profile.py

```diff
@@ -575,3 +578,52 @@ def _is_memory_only(nm_profile):
             or NM.SettingsConnectionFlags.VOLATILE & profile_flags
         )
     return False
+
+
+_NM_TYPE_2_NMSTATE = {
+    NM.SETTING_WIRED_SETTING_NAME: InterfaceType.ETHERNET,
+    NM.SETTING_BOND_SETTING_NAME: InterfaceType.BOND,
+    NM.SETTING_DUMMY_SETTING_NAME: InterfaceType.DUMMY,
+    NM.SETTING_TEAM_SETTING_NAME: InterfaceType.TEAM,
+    NM.SETTING_VLAN_SETTING_NAME: InterfaceType.VLAN,
+    NM.SETTING_VXLAN_SETTING_NAME: InterfaceType.VXLAN,
+    NM.SETTING_BRIDGE_SETTING_NAME: InterfaceType.LINUX_BRIDGE,
+    NM.SETTING_VRF_SETTING_NAME: InterfaceType.VRF,
+    NM.SETTING_INFINIBAND_SETTING_NAME: InterfaceType.INFINIBAND,
+    NM.SETTING_MACVLAN_SETTING_NAME: InterfaceType.MAC_VLAN,
+    NM.SETTING_OVS_BRIDGE_SETTING_NAME: InterfaceType.OVS_BRIDGE,
+    NM.SETTING_OVS_PORT_SETTING_NAME: InterfaceType.OVS_PORT,
+    NM.SETTING_OVS_INTERFACE_SETTING_NAME: InterfaceType.OVS_INTERFACE,
+    NM.SETTING_VETH_SETTING_NAME: InterfaceType.VETH,
```

Comment:
In order to provide backward compatibility `NM.SETTING_VETH_SETTING_NAME` cannot be called directly. A private helper function it is needed as done in `libnmstate/nm/translator.py` and `libnmstate/nm/veth.py`